BHoM_Engine/Modify/AddFragment.cs

```diff
@@ -44,10 +44,14 @@ public static IBHoMObject AddFragment(this IBHoMObject iBHoMObject, IBHoMFragmen
             if (iBHoMObject == null) return null;
             IBHoMObject o = iBHoMObject.DeepClone();
             o.Fragments = new List<IBHoMFragment>(iBHoMObject.Fragments);
-            if(o.Fragments.Contains(fragment))
+
+            if(o.Fragments.Where(x => x.GetType() == fragment.GetType()).FirstOrDefault() != null)
```

Comment:
Not the biggest fan of this. As a minimum, I would rather use:

```
o.Fragments.Find(x => x.GetType() == fragment.GetType()) != null
```

But, given what is coming below, I would prefer

```
int index = o.Fragments.FindIndex(x => x.GetType() == fragment.GetType());
if (index >= 0)
{
    if(replace)
        o.Fragments[index] = fragment;
    else
        ...
}
```

Reply:
Changes made :smile: